iceoryx_binding_c/include/iceoryx_binding_c/wait_set.h

```diff
@@ -35,56 +35,38 @@ iox_ws_t iox_ws_init(iox_ws_storage_t* self);
 /// @param[in] self handle to the wait set
 /// @param[in] timeout duration how long this method should wait
-/// @param[in] conditionArray preallocated memory to an array of iox_cond_t in which
-///             the conditions which were triggered can be written to
-/// @param[in] conditionArrayCapacity the capacity of the preallocated conditionArray
-/// @param[in] missedElements if the conditionArray has insufficient size the number of missed elements
+/// @param[in] triggerArray preallocated memory to an array of iox_cond_t in which
+///             the triggers which were triggered can be written to
+/// @param[in] triggerArrayCapacity the capacity of the preallocated triggerArray
+/// @param[in] missedElements if the triggerArray has insufficient size the number of missed elements
 ///             which could not be written into the array are stored here
-/// @return number of elements which were written into the conditionArray
+/// @return number of elements which were written into the triggerArray
 uint64_t iox_ws_timed_wait(iox_ws_t const self,
                            struct timespec timeout,
-                           iox_cond_t* const conditionArray,
-                           const uint64_t conditionArrayCapacity,
+                           iox_trigger_state_t const triggerArray,
+                           const uint64_t triggerArrayCapacity,
                            uint64_t* missedElements);
 
-/// @brief waits until a condition was triggered
+/// @brief waits until a trigger was triggered
 /// @param[in] self handle to the wait set
-/// @param[in] conditionArray preallocated memory to an array of iox_cond_t in which
-///             the conditions which were triggered can be written to
-/// @param[in] conditionArrayCapacity the capacity of the preallocated conditionArray
-/// @param[in] missedElements if the conditionArray has insufficient size the number of missed elements
+/// @param[in] triggerArray preallocated memory to an array of iox_cond_t in which
+///             the triggers which were triggered can be written to
+/// @param[in] triggerArrayCapacity the capacity of the preallocated triggerArray
+/// @param[in] missedElements if the triggerArray has insufficient size the number of missed elements
 ///             which could not be written into the array are stored here
-/// @return number of elements which were written into the conditionArray
+/// @return number of elements which were written into the triggerArray
 uint64_t iox_ws_wait(iox_ws_t const self,
-                     iox_cond_t* const conditionArray,
-                     const uint64_t conditionArrayCapacity,
+                     iox_trigger_state_t const triggerArray,
+                     const uint64_t triggerArrayCapacity,
                      uint64_t* missedElements);
 
+/// @brief returns the amount of acquired triggers
+uint64_t iox_ws_size(iox_ws_t const self);
+
+/// @brief returns the overall capacity of the waitset
```

Comment:
capacity means how many triggers could be acquired? Maybe better name and description. Also for `iox_ws_size`

Reply:
But it is common to use `capacity` and `size` in exactly that context. See: https://en.cppreference.com/w/cpp/container/vector, capacity of a vector quote: "returns the number of elements that can be held in currently allocated storage", size: "returns the number of elements"